### PR TITLE
Merge ipc/tcp codecs to one, configurable

### PR DESCRIPTION
**Pull request by NikVolf**

### CODE REVIEW

File: server-utils/src/lib.rs [hunk 1]
```diff
@@ -15,5 +17,12 @@ pub mod hosts;
 pub mod session;
 pub mod reactor;
 mod matcher;
+mod stream_codec;
 
 pub use matcher::Pattern;
+
+/// Codecs utilities
+pub mod codecs {
```
Comment (tomusdrw) [inline server-utils/src/lib.rs]: Why not just make `stream_codec` module public?

File: server-utils/src/stream_codec.rs [hunk 2]
```diff
@@ -1,8 +1,56 @@
-use std::io;
-use server_utils::tokio_io::codec::{Decoder, Encoder};
+use std::{io, str};
+use tokio_io::codec::{Decoder, Encoder};
 use bytes::BytesMut;
 
-pub struct StreamCodec;
+/// Separator for enveloping messages in streaming codecs
+pub struct Separator(Option<u8>);
```
Comment (tomusdrw) [inline server-utils/src/stream_codec.rs]: I think `Option`-like enum would do here, why do you need a separate struct that wraps it?

File: server-utils/src/lib.rs (hunk 1 above)
Comment (NikVolf) [inline server-utils/src/lib.rs]: Just wanted all codecs (currently only one) it accessible via short namespace `codecs`

File: server-utils/src/stream_codec.rs (hunk 2 above)
Comment (NikVolf) [inline server-utils/src/stream_codec.rs]: so i could make a separate `impl Default` for it
along with other custom methods

Comment (NikVolf) [inline server-utils/src/stream_codec.rs]: oh you probably ment smth like 

```
enum Separator { Empty, Byte(u8) }
```
?

that makes sense

Comment (tomusdrw) [inline server-utils/src/stream_codec.rs]: Yeah.

File: server-utils/src/lib.rs (hunk 1 above)
Comment (tomusdrw) [inline server-utils/src/lib.rs]: k

File: server-utils/src/stream_codec.rs (hunk 2 above)
Comment (NikVolf) [inline server-utils/src/stream_codec.rs]: Updated!

**Pull request merged**